src/web/sitemap.rs

```diff
@@ -64,37 +65,39 @@ pub(crate) async fn sitemap_handler(
             return Err(AxumNope::ResourceNotFound);
         }
     }
-    let releases = spawn_blocking(move || {
-        let mut conn = pool.get()?;
-        let query = conn.query(
-            "SELECT crates.name,
-                    releases.target_name,
-                    MAX(releases.release_time) as release_time
-             FROM crates
-             INNER JOIN releases ON releases.crate_id = crates.id
-             WHERE
-                rustdoc_status = true AND
-                crates.name ILIKE $1
-             GROUP BY crates.name, releases.target_name
-             ",
-            &[&format!("{letter}%")],
-        )?;
-
-        Ok(query
-            .into_iter()
-            .map(|row| SitemapRow {
-                crate_name: row.get("name"),
-                target_name: row.get("target_name"),
-                last_modified: row
-                    .get::<_, DateTime<Utc>>("release_time")
-                    // On Aug 27 2022 we added `<link rel="canonical">` to all pages,
-                    // so they should all get recrawled if they haven't been since then.
+
+    let mut conn = pool.get_async().await?;
+
+    let releases: Vec<_> = sqlx::query!(
+        "SELECT crates.name,
+                releases.target_name,
+                MAX(releases.release_time) as release_time
```

Comment:
```suggestion
                MAX(releases.release_time) as "release_time!"
```

This is non-null, sqlx's inference just can't see through the `MAX` call. (Which will remove the `.map().unwrap_or_default()` below too).